packages/main/src/Tree.ts

```diff
@@ -0,0 +1,515 @@
+	 *
+	 * @protected
+	 * @since 1.0.0-rc.8
+	 */
+	@property({ type: Boolean })
+	_minimal!: boolean;
+
+	/**
+	 * Defines the items of the component. Tree items may have other tree items as children.
+	 * <br><br>
+	 * <b>Note:</b> Use <code>ui5-tree-item</code> for the intended design.
+	 *
+	 * @type {sap.ui.webc.main.ITreeItem[]}
+	 * @name sap.ui.webc.main.Tree.prototype.default
+	 * @slot items
+	 * @public
+	 */
+	@slot({ type: HTMLElement, invalidateOnChildChange: true, "default": true })
+	items!: Array<TreeItemBase>;
+
+	/**
+	 * Defines the component header.
+	 * <br><br>
+	 * <b>Note:</b> When the <code>header</code> slot is set, the
+	 * <code>headerText</code> property is ignored.
+	 *
+	 * @type {HTMLElement[]}
+	 * @name sap.ui.webc.main.Tree.prototype.header
+	 * @slot header
+	 * @public
+	 */
+	@slot()
+	header!: Array<HTMLElement>;
+
+	static get render() {
+		return litRender;
+	}
+
+	static get styles() {
+		return TreeCss;
+	}
+
+	static get template() {
+		return TreeTemplate;
+	}
+
+	static get dependencies() {
+		return [
+			TreeList,
+			TreeItem,
+			TreeItemCustom,
+		];
+	}
+
+	onBeforeRendering() {
+		this._prepareTreeItems();
+	}
+
+	get list() {
+		return this.getDomRef() as TreeList;
+	}
+
+	get _role() {
+		return this._minimal ? "menubar" : "tree";
+	}
+
+	get _label() {
+		return getEffectiveAriaLabelText(this);
+	}
+
+	_onListItemStepIn(e: CustomEvent<TreeItemBaseStepInEventDetail>) {
+		const treeItem = e.detail.item;
+		if (treeItem.items.length > 0) {
+			const firstChild = treeItem.items[0];
+			const firstChildListItem = this._getListItemForTreeItem(firstChild);
+			firstChildListItem && this.list.focusItem(firstChildListItem);
+		}
+	}
+
+	_onListItemStepOut(e: CustomEvent<TreeItemBaseStepOutEventDetail>) {
+		const treeItem = e.detail.item;
+		if (treeItem.parentElement !== this) {
+			const parent = treeItem.parentElement as TreeItemBase;
+			const parentListItem = this._getListItemForTreeItem(parent);
+			parentListItem && this.list.focusItem(parentListItem);
+		}
+	}
+
+	_onListItemToggle(e: CustomEvent<TreeItemBaseToggleEventDetail>) {
+		const treeItem = e.detail.item;
+		const defaultPrevented = !this.fireEvent<TreeItemToggleEventDetail>("item-toggle", { item: treeItem }, true);
+		if (!defaultPrevented) {
+			treeItem.toggle();
+		}
+	}
+
+	_onListItemClick(e: CustomEvent<ListItemClickEventDetail>) {
+		const treeItem = e.detail.item as TreeItemBase;
+
+		if (!this.fireEvent<TreeItemClickEventDetail>("item-click", { item: treeItem }, true)) {
+			e.preventDefault();
+		}
+	}
+
+	_onListItemDelete(e: CustomEvent<ListItemDeleteEventDetail>) {
+		const treeItem = e.detail.item as TreeItemBase;
+		this.fireEvent<TreeItemDeleteEventDetail>("item-delete", { item: treeItem });
+	}
+
+	_onListItemMouseOver(e: MouseEvent) {
+		const target = e.target,
+			isInstanceOfTreeItemBase = (object: any): object is TreeItemBase => {
```

Comment:
you can define isInstanceOfTreeItemBase as local function inside module and to reuse it in _onListItemMouseOut. When the function is declared to the module you can replace `(target as any).isTreeItem` with `isInstanceOfTreeItemBase(target)`

Reply:
Oops, forgot the second usage... thnx, done